source/common/router/router.cc

```diff
@@ -837,6 +849,37 @@ void Filter::sendNoHealthyUpstreamResponse() {
                              StreamInfo::ResponseCodeDetails::get().NoHealthyUpstream);
 }
 
+void Filter::giveUpRetryAndShadow() {
+  cluster_->trafficStats()->retry_or_shadow_abandoned_.inc();
+  retry_state_.reset();
+  active_shadow_policies_.clear();
+  request_buffer_overflowed_ = true;
+  if (callbacks_->decodingBuffer() != nullptr) {
+    // We previously went over the buffer limit, which pause the read on the downstream connection.
+    // draining the buffer so we enable read on the downstream connection.
+    // This buffer is only used to send the body when retrying and shadowing which has been disable
+    // above.
+    callbacks_->modifyDecodingBuffer([](Buffer::Instance& data) { data.drain(data.length()); });
```

Comment:
not sure we should actually drain the buffer as it might me used by some other filters.
But then should we keep the HCM buffer limit the new values

Reply:
removed it, this was making quic and mobile test fail